### ISSUE TITLE
default query filters

### ISSUE THREAD

**Issue by backwardspy** [piccolo-orm/piccolo#657]
hi!

it would be useful to me to be able to somehow specify a default set of `.where(...)` clauses to include on a query on my tables.

in my case, i'm soft-deleting objects by setting a flag that prevents the objects from coming back in queries. of course this means I have to remember to add a clause for that field in all of my queries, and if I forget one then soft-deleted objects start appearing in my API responses.

i think in django it's possible to do this with [managers](https://docs.djangoproject.com/en/4.1/topics/db/managers/). for example you could have `Band.objects().where(...)` that allows you to query for non-deleted bands only, and a `Band.all_objects().where(...)` that bypasses this and allows queries over the whole table.

please let me know what you think about this functionality, and if there's already some way to do this that i missed.

**Comment by dantownsend** [comment]
Yeah, we could potentially add something to support this. Not sure on the perfect API though. What do you think?

These are the sorts of work arounds I use at the moment:

```python

class Band(Table):
    deleted = Boolean(default=False)

    @classmethod
    def non_deleted(cls):
        return cls.select().where(cls.deleted == False)

# And then you can chain extra methods onto it:
>>> await Band.non_deleted().where(...)
```

Or maybe this:

```python
class Band(Table):
    deleted = Boolean(default=False)

    @classmethod
    def select(cls, *args, **kwargs):
        return super().select(*args, **kwargs).where(cls.deleted == False)
```

**Comment by backwardspy** [comment]
that's not bad to be honest, at the very least it suits my use case at the moment.

can you think of anything that "native" support for this would offer that a classmethod does not?

**Comment by dantownsend** [comment]
Maybe for some use cases a dedicated API is required. I tend to use custom class methods quite a lot. In the past when I used Django, I used class methods instead of custom managers. I know some people preferred the custom manager approach though.
